File: oldapp/src/main/java/com/riprog/launcher/LauncherApplication.java
package com.riprog.launcher;

import android.app.Application;

public class LauncherApplication extends Application {
    private LauncherModel model;

    @Override
    public void onCreate() {
        super.onCreate();
        model = new LauncherModel(this);
    }

    public LauncherModel getModel() {
        return model;
    }

    @Override
    public void onTrimMemory(int level) {
        super.onTrimMemory(level);
        if (model != null) {
            model.onTrimMemory(level);
        }
    }

    @Override
    public void onTerminate() {
        super.onTerminate();
        if (model != null) {
            model.shutdown();
        }
    }
}

File: oldapp/src/main/java/com/riprog/launcher/LauncherApplication.kt
package com.riprog.launcher

import com.riprog.launcher.data.repository.AppRepository

import android.app.Application

class LauncherApplication : Application() {
    lateinit var model: AppRepository
        private set

    override fun onCreate() {
        super.onCreate()
        model = AppRepository(this)
    }

    override fun onTrimMemory(level: Int) {
        super.onTrimMemory(level)
        model.onTrimMemory(level)
    }

    override fun onTerminate() {
        super.onTerminate()
        model.shutdown()
    }
}
